src/other/tables.jl

```diff
@@ -16,10 +16,13 @@ fromcolumns(x; copycols::Bool=true) =
               Index(collect(Symbol, propertynames(x))),
               copycols=copycols)
 
-function DataFrame(x; copycols::Bool=true)
+function DataFrame(x::T; copycols::Bool=true) where {T}
     if x isa AbstractVector && all(col -> isa(col, AbstractVector), x)
         return DataFrame(Vector{AbstractVector}(x), copycols=copycols)
     end
+    if Tables.istable(T)
```

Comment:
if we add this I understand we should remove the code starting in line 32 below (https://github.com/JuliaData/DataFrames.jl/pull/1789/files#diff-b84fca182aa841ee4af16b7d5d6e3399R32)

Reply:
No, they do 2 different things. It's subtle, but here we're checking if the ***type*** of `T` `istable`, and if so, dispatching directly to Tables.jl. Below, we check if the ***instance*** `istable`, which is currently much more liberal (though there is a [PR](https://github.com/JuliaData/Tables.jl/pull/84) to make it conservative again). The idea here is that most official Tables.jl-compatible sources implement `Tables.istable(::Type{T}) = true` directly, so we can rely on that to dispatch directly to Tables.jl before needing to do the `iterate` thing.

Reply:
OK - thank you.